tests/analyze_tests.py

```diff
@@ -0,0 +1,53 @@
+from pathlib import Path
+import yaml
+
+template = '''
+pytest.{name}:
+  extends: .pytest-{extends}
+  variables:
+    PYTESTFILE: {test_file}
+  allow_failure: {allow_failure}
+'''
+
+# override the auto detection of which script to extend for the following jobs
+extends_override = {'backends' : 'fpga'}
+
+# allow the following jobs to fail
+allow_failure = ['backends', 'xgb_converter', 'onnx_to_hls']
+
+# check whether "build" method is called in the test -> needs different resources
```

Comment:
Does there need to be a similar method for seeing if xvhdl is needed in the test and allocate resources accordingly?

Reply:
That's what the "extends_override" is for, but it looks like I missed quite a few